python/kernmlops/kernmlops_benchmark/redis.py

```diff
@@ -95,6 +97,15 @@ def run(self) -> None:
         if self.server is not None:
             raise BenchmarkRunningError()
 
+        if self.config.mem_cgroup_size:
+            cgroup_path = "/sys/fs/cgroup/redis_mem"
+            subprocess.run(["mkdir", "-p", cgroup_path], check=True)
+            with open(f"{cgroup_path}/memory.max", "w") as f:
+                f.write(str(self.config.mem_cgroup_size))
+            # redis inherits cgroup from current process
+            with open(f"{cgroup_path}/cgroup.procs", "w") as f:
+                f.write(str(os.getpid()))
```

Comment:
shouldn't you be able to do this with a launching pre_exec function in subprocess?